SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -0,0 +1,245 @@
+          new CPolynomialViscosity<N_POLY_COEFFS>(config->GetMu_PolyCoeffND()));
+      }
+      break;
+    case VISCOSITYMODEL::FLAMELET:
+      /* Do nothing. Viscosity is obtained from the table and set in SetTDState_T */
+      break;
+    default:
+      SU2_MPI::Error("Viscosity model not available.", CURRENT_FUNCTION);
+      break;
+  }
+}
+
+void CFluidScalar::SetThermalConductivityModel(const CConfig* config) {
+  switch (config->GetKind_ConductivityModel()) {
+    case CONDUCTIVITYMODEL::CONSTANT:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantConductivityRANS>(
+            new CConstantConductivityRANS(config->GetThermal_Conductivity_Constant(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantConductivity>(new CConstantConductivity(config->GetThermal_Conductivity_Constant(iVar)));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::CONSTANT_PRANDTL:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantPrandtlRANS>(
+            new CConstantPrandtlRANS(config->GetPrandtl_Lam(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantPrandtl>(new CConstantPrandtl(config->GetPrandtl_Lam(iVar)));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::POLYNOMIAL:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivity = std::unique_ptr<CPolynomialConductivityRANS<N_POLY_COEFFS>>(
+            new CPolynomialConductivityRANS<N_POLY_COEFFS>(config->GetKt_PolyCoeffND(), config->GetPrandtl_Turb()));
+        } else {
+          ThermalConductivity = std::unique_ptr<CPolynomialConductivity<N_POLY_COEFFS>>(
+            new CPolynomialConductivity<N_POLY_COEFFS>(config->GetKt_PolyCoeffND()));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::FLAMELET:
+      /* Do nothing. Conductivity is obtained from the table and set in SetTDState_T */
+      break;
+    default:
+      SU2_MPI::Error("Conductivity model not available.", CURRENT_FUNCTION);
+      break;
+  }
+}
+
+std::vector<su2double>& CFluidScalar::massToMoleFractions(const su2double * const val_scalars){
+  su2double mixtureMolarMass {0.0};
+
+  /* Change if val_scalars becomes array of scalar su2double*. */
+  massFractions[0] = val_scalars[0];
+  massFractions[1] = 1 - val_scalars[0];
+
+  for(int iVar = 0; iVar < n_species_mixture; iVar++){
+    mixtureMolarMass += massFractions[iVar] / molarMasses[iVar];
+  }
```

Comment:
The top bit is hardcoded for `n_species_mixture=2` overall species, and the bottom for arbitrary many. Lets keeps in mind that at some point I think it is better to keep it open for arbitrary many species

Reply:
Yes I agree, I will change that somewhere down the road.